src/model_theory/basic.lean

```diff
@@ -59,19 +59,36 @@ structure language :=
 (functions : ℕ → Type u) (relations : ℕ → Type v)
 
 /-- Used to define `first_order.language₂`. -/
-def sequence₂ (a₀ a₁ a₂ : Type u) : ℕ → Type u
+@[simp] def sequence₂ (a₀ a₁ a₂ : Type u) : ℕ → Type u
 | 0 := a₀
 | 1 := a₁
 | 2 := a₂
 | _ := pempty
 
 instance {a₀ a₁ a₂ : Type u} [h : inhabited a₀] : inhabited (sequence₂ a₀ a₁ a₂ 0) := h
 
+instance {a₀ a₁ a₂ : Type u} {n : ℕ} : is_empty (sequence₂ a₀ a₁ a₂ (n + 3)) := pempty.is_empty
+
+@[simp] lemma sum_card_sequence₂ (a₀ a₁ a₂ : Type u) :
+  cardinal.sum (λ i, # (sequence₂ a₀ a₁ a₂ i)) = # a₀ + # a₁ + # a₂ :=
+begin
+  rw [sum_nat_eq_add_sum_succ, sum_nat_eq_add_sum_succ, sum_nat_eq_add_sum_succ],
+  simp [add_assoc],
+end
+
+@[simp] lemma sum_lift_card_sequence₂ (a₀ a₁ a₂ : Type u) :
```

Comment:
If you prove this first, then `simp` will be able to prove `sum_card_sequence₂`.

Reply:
When I try that, `simp` doesn't make any progress on `sum_card_sequence2`.

Reply:
I could deduce one from the other if I show that `sequence₂` commutes in some way with `lift` and `ulift`, but I'm not sure that's worth it for these short proofs, as I don't particularly expect it'll show up elsewhere.

Reply:
Well, maybe it will pop up if we ever need to lift languages to particular universes, so here it is.